pkg/kubelet/types/types.go

```diff
@@ -68,6 +68,22 @@ func (s SortedContainerStatuses) Less(i, j int) bool {
 	return s[i].Name < s[j].Name
 }
 
+// SortInitContainerStatuses ensures that statuses are in the order that their
+// init container appears in the pod spec
+func SortInitContainerStatuses(p *api.Pod, statuses []api.ContainerStatus) {
+	containers := p.Spec.InitContainers
+	current := 0
+	for _, container := range containers {
+		for j := current; j < len(statuses); j++ {
```

Comment:
Hrm, if someone passes a statuses where container.Name is not in statuses, will this loop forever?